### PR TITLE
Updated DynamicApis to not use reflection on platforms where the APIs are available

### LINKED ISSUE #26: PublishTrimmed cause XML documentation to disappear
In .NET core >= 3.0, when `<PublishTrimmed>true</PublishTrimmed>` and the project that uses this package is published, the following libraries get trimmed, because reflection is used:

- System.IO.FileSystem
- System.Xml.XPath.XDocument
- System.Xml.Linq

# Impact
XML documentation cannot be loaded by this library. This can affect NSwag, where the generated swagger.json lacks documentation for published applications with this flag enabled. This is not blocking as the application works normally, but it's a difference between the application tested in the dev environment and the application that gets pushed into production.

# Workaround
You can use one of the technique described [here](https://github.com/mono/linker/blob/master/src/ILLink.Tasks/README.md#adding-reflection-roots).
For example, you could add these lines into your .csproj:

```xml
  <ItemGroup>
    <TrimmerRootAssembly Include="System.IO.FileSystem" />
    <TrimmerRootAssembly Include="System.Xml.XPath.XDocument" />
    <TrimmerRootAssembly Include="System.Xml.Linq" />
  </ItemGroup>
```

# What are the options for Namotion.Reflection / NSwag
- Do nothing : People that gets those issues will be redirected here after a quick google search
- Warn users clearly in the documentation : "This package makes use of reflection, if you are using PublishTrimmed in your project, please see ..."
- Explicitely force users to include the `TrimmerRootAssembly` by adding a `build/<PackageName>.props` into Namotion.Reflection's package, which contains the above XML inside a `<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">` tag.
In N.R.csproj, add this so that it gets packed:
```xml
  <ItemGroup>
    <None Include="build\Namotion.Reflection.props" Pack="True" PackagePath="build\" />
  </ItemGroup>
```
Direct references to this package would then have the TrimmerRootAssembly added, and these assemblies should not get trimmed.

For indirect references (like in NJS.csproj), you can use:
```xml
<PackageReference Include="Namotion.Reflection" Version="XXX" PrivateAssets="None" />
```

The value of `None` might be overkill here, see the doc. The idea is to tell msbuild "Let properties/item groups leak into my project, and get referenced as well when someone references me"
- Remove the use of Dynamically loaded APIs. You can do that by keeping the code between
```cs
#if NETSTANDARD1_0
// Dynamically loaded API
#else
// API is always present, and use it directly
#endif
```

# Impacted target frameworks
Netstandard & netcoreapp that get included in a netcoreapp application.

### PR DESCRIPTION
**Pull request by jeremyVignelles**
For compatibility, all platforms are currently forced to use reflection to do simple tasks like File.Exists, only because netstandard1.0 doesn't support this.

The proposed change does not change the behavior for netstandard1.0, but allows other TFMs to use the APIs directly, because they are available.

I expect this change to improve performance (not benchmarked), because DynamicApis does not use reflection anymore on newer platforms.

This should also fix #26 because required types are actually referenced, and I don't think that they would get trimmed anymore.

### CODE REVIEW

File: src/Namotion.Reflection/Infrastructure/DynamicApis.cs [hunk 1]
```diff
@@ -42,16 +48,33 @@ static DynamicApis()
 
         /// <summary>Gets a value indicating whether XPath APIs are available.</summary>
         public static bool SupportsXPathApis => XPathExtensionsType != null;
+#else
+        /// <summary>Gets a value indicating whether file APIs are available.</summary>
+        public static bool SupportsFileApis => true;
```
Comment (RicoSuter) [inline src/Namotion.Reflection/Infrastructure/DynamicApis.cs]: Keep the properties and always return true to have a common public api and avoid breaking changes?

Comment (jeremyVignelles) [reply src/Namotion.Reflection/Infrastructure/DynamicApis.cs]: Yes, that was the goal, not breaking anything. I don't know the project, and I don't know the implication of a change of the API here

Comment (RicoSuter) [reply src/Namotion.Reflection/Infrastructure/DynamicApis.cs]: Ah sorry it is already like that - long day

**Comment by RicoSuter** [comment]
Lgtm, but i wonder why i didnt do it this way in the first place... 🙂

**Comment by RicoSuter** [comment]
I cannot remember and today it makes absolutely sense to do it this way. 
Thanks for the PR.

**Comment by RicoSuter** [comment]
BTW: Performance was never a problem because there was max. one XML file read per DLL, and reading the DLL, doing all the reflection probably takes much more time then doing this single dynamic File API call.

**Pull request merged**